### ISSUE TITLE
Possible bug in pls_predict_scores() for RKHS (kernel PLS) fits

### ISSUE THREAD

**Issue by gloignon** [fbertran/bigPLSR#1]
Bonjour!

I am using bigPLSR version 0.7.2 with my own cross-validation setup. I fit with pls_fit(algorithm = "rkhs") and score held-out rows with pls_predict_scores(). I noticed that the score changes depending on what other data is in the test set. It appears that for centering the function uses the mean from the data we are scoring.

Expected behavior for my setup: new rows are centered using the training set (or data we used to "train" the model) means, so when we apply the model each row gets the same score regardless of the other rows that are in its batch.

Regards,
GL

Example:
```
> library(bigPLSR)
> set.seed(1)
> X <- matrix(rnorm(200 * 10), 200)
> y <- X[, 1] + sin(2 * X[, 2]) + rnorm(200, sd = 0.2)
> fit <- pls_fit(X[1:150, ], y[1:150], ncomp = 2, algorithm = "rkhs",
+                kernel = "rbf", gamma = 0.05, scores = "r")
> batch_a <- X[c(151, 152:161), ]
> batch_b <- X[c(151, 162:171), ]
> pls_predict_scores(fit, batch_a)[1, ]  # row 151 scored with rows 152-161
        t1         t2 
 0.1177865 -0.1280255 
> pls_predict_scores(fit, batch_b)[1, ]  # row 151 scored with rows 162-171
         t1          t2 
 0.06585747 -0.13649166 
```
